### PR TITLE
Updated documentation for create/edit view

### PR DESCRIPTION
**Pull request by TimoBakx**
I am targeting the 3.x branch, because these are updates to the documentation of that branch.

Updates #1557

Please let me know if I'm doing this correctly or if I should change anything. This pull request does not complete #1557 fully, but it's a step.

## Subject

Improved documentation of the create/edit view.

### CODE REVIEW

File: Resources/doc/reference/action_create_edit.rst [hunk 1]
```diff
@@ -36,11 +36,34 @@ SonataAdmin Options that may affect the create or edit view:
...
+~~~~~~
+
+You can disable creating or editing objects by removing the corresponding routes in your Admin.
+
+.. code-block:: php
+
+    /**
+     * @param RouteCollection $collection
+     * @return void
+     */
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: You may remove this, assume php 7 and add `: void` at the end of line 54

File: Resources/doc/reference/action_create_edit.rst [hunk 2]
```diff
@@ -60,6 +60,7 @@ You can disable creating or editing objects by removing the corresponding routes
 
         /* Removing the create route will disable creating new entities. It will also
          * remove the 'Add new' button in the list view.
+         */
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Please fixup this commit with the previous one

File: Resources/doc/reference/action_create_edit.rst [hunk 3]
```diff
@@ -103,6 +113,9 @@ To specify options, do as follows:
                     ->end()
                 ->end()
             ;
+        }
+
+        // ...
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Please also fixup this commit, the diff will be less confusing

File: Resources/doc/reference/action_create_edit.rst [hunk 4]
```diff
@@ -74,6 +74,53 @@ You can disable creating or editing objects by removing the corresponding routes
         // ...
     }
 
+Adding form fields
+~~~~~~~~~~~~~~~~~~
+
+Within the configureFormFields method you can define which fields should be shown when editing or creating entities.
+Each field has to be added to a specific form group. And form groups can optionally be added to a tab. See `FormGroup options`_ for aditional information about configuring form groups.
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Please comply with the sf docs standards:

> [Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);](http://symfony.com/doc/current/contributing/documentation/standards.html)

Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: "aditional" => "additional"

File: Resources/doc/reference/action_create_edit.rst [hunk 5]
```diff
@@ -74,6 +74,53 @@ You can disable creating or editing objects by removing the corresponding routes
...
+
+Using the FormMapper add method, you can add form fields. The add method has 3 attributes:
+- name (the name of your entity)
+- type (defaults to null to let Sonata decide which type to use)
+- options (see `Field Types <field_types.rst>` for more information)
+
+.. note::
+
+    The attribute name entered in name should be available in your Entity through public access or getter/setter.
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: "attribute name" => "property" might be less confusing?

File: Resources/doc/reference/action_create_edit.rst [hunk 6]
```diff
@@ -74,6 +74,53 @@ You can disable creating or editing objects by removing the corresponding routes
...
+         * @return void
+         * @throws \RuntimeException
+         */
+        protected function configureFormFields(FormMapper $formMapper)
+        {
+            $formMapper
+                ->tab('General') // The tab call is optional
+                    ->with('Addresses')
+                        ->add('title') // Add a field and let Sonata decide which type to use
+                        ->add('streetname', 'text') // Add a textfield
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: I think the recommended way now is `TextType::class`, isn't it?

File: Resources/doc/reference/action_create_edit.rst [hunk 7]
```diff
@@ -83,7 +83,7 @@ Each field has to be added to a specific form group. And form groups can optiona
 Using the FormMapper add method, you can add form fields. The add method has 3 attributes:
 - name (the name of your entity)
 - type (defaults to null to let Sonata decide which type to use)
-- options (see `Field Types <field_types.rst>` for more information)
+- options (see `Field Types <field_types.rst>`_ for more information)
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Please fixup this commit with the offending commit

File: Resources/doc/reference/action_create_edit.rst [hunk 8]
```diff
@@ -80,10 +80,11 @@ Adding form fields
...
 
-Using the FormMapper add method, you can add form fields. The add method has 3 attributes:
-- name (the name of your entity)
-- type (defaults to null to let Sonata decide which type to use)
-- options (see `Field Types <field_types.rst>`_ for more information)
+Using the FormMapper add method, you can add form fields. The add method has 3 parameters:
+
+- ``name``: the name of your entity
+- ``type``: defaults to null to let Sonata decide which type to use
+- ``options``: see `Field Types <field_types.rst>`_ for more information
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Please fixup this commit with the commit that introduced the list.

**Comment by greg0ire** [comment]
As every PR, this is blocked by https://github.com/sonata-project/dev-kit/pull/267

File: Resources/doc/reference/action_create_edit.rst (hunk 6 above)
Comment (core23) [inline Resources/doc/reference/action_create_edit.rst]: Same for the other types like `integer`, ...

File: Resources/doc/reference/action_create_edit.rst [hunk 9]
```diff
@@ -14,26 +14,132 @@ of the fields and forms available in these views and any other relevant settings
...
 
+SonataAdmin Options that may affect the create or edit view:
+
+.. code-block:: yaml
+
+    sonata_admin:
+        options:
+            html5_validate:       true     # enable or disable html5 form validation
+            confirm_exit:         true     # enable or disable a confirmation before navigating away
+            use_select2:          true     # enable or disable usage of the Select2 jQuery library (https://github.com/select2/select2)
```
Comment (greg0ire) [inline Resources/doc/reference/action_create_edit.rst]: Can you please wrap the long comments here? They make an horizontal scroll appear.

**Comment by Xymanek** [comment]
For the routing section I think it would make sense to add link to more detailed page (https://sonata-project.org/bundles/admin/3-x/doc/reference/routing.html)

**Comment by TimoBakx** [comment]
@Xymanek Good point. I added the reference link.

**Comment by OskarStark** [comment]
Please rebase your PR, we fixed the travis configuration for PHP 5.3 builds, thank you!

**Comment by greg0ire** [comment]
No need to rebase @Oskarstark ;)

**Comment by TimoBakx** [comment]
@greg0ire @OskarStark Too late...

**Comment by OskarStark** [comment]
> No need to rebase @OskarStark ;)

Oh you are right, my fault 👍

**Comment by OskarStark** [comment]
Thank you @TimoBakx 👍

**Comment by TimoBakx** [comment]
Thanks for the reviews, feedback and help @greg0ire, @Xymanek, @core23 and @OskarStark.

**Pull request merged**